bmi270: Allow interrupt handling for spi too

electrocnic opened the pull request:
Interrupt handling is independent from the used BUS (i2c or spi) and therefore should be in the parent binding file instead of a bus-specific binding file. The interrupt code is handling interrupts already independently from the used BUS. See also this ticket for more information: https://github.com/zephyrproject-rtos/zephyr/issues/58843

galak commented:
@electrocnic please fix the CI complaince check issue.

electrocnic commented:
> @electrocnic please fix the CI complaince check issue.

done.

MaureenHelm reviewed (dismissed):
Thanks for the fix! Can you look into adding this property to the bmi270 nodes in `tests/drivers/build_all/sensor/spi.dtsi` and `tests/drivers/build_all/sensor/i2c.dtsi`

electrocnic commented:
> Thanks for the fix! Can you look into adding this property to the bmi270 nodes in `tests/drivers/build_all/sensor/spi.dtsi` and `tests/drivers/build_all/sensor/i2c.dtsi`

Thanks for the hint.
I hope it is okay to format some other lines in these files with my commit?

MaureenHelm commented:
> > Thanks for the fix! Can you look into adding this property to the bmi270 nodes in `tests/drivers/build_all/sensor/spi.dtsi` and `tests/drivers/build_all/sensor/i2c.dtsi`
> 
> Thanks for the hint. I hope it is okay to format some other lines in these files with my commit?

Ok to do in the same PR, but need to be in a separate commit.

electrocnic committed:
dts: sensor: bosch,bmi270: Move interrupt handling to parent binding

Moved the interrupt handling code from a bus-specific binding file to
the parent binding file. This modification ensures that interrupt
handling, which is inherently independent of the type of BUS used
(either i2c or spi), is located in the appropriate location within
the code.

Previously, this code was found in a bus-specific file, despite its
functionality being applicable to all buses. This change ensures a
more logical placement and will help to maintain coherence within the
codebase.

This change aligns with the existing implementation where the interrupt
handling code already operates independently of the BUS type.

Tested this on a nrf52840 with a bmi270 on spi bus with the sample from
zephyr/samples/sensor/bmi270 by adding an interrupt handler in main.c
which uses the bmi270_trigger.c implementation and verified this with
breakpoints and log output.

Added the irq-gpios to tests/drivers/build_all/sensor/i2c.dtsi and
tests/drivers/build_all/sensor/spi.dtsi and fixed whitespace formatting.

Fixes: #58843

Signed-off-by: Andreas Wiesinger <awiesing90@gmail.com>

electrocnic commented:
@MaureenHelm @galak @avisconti @teburd

I hate to say this, but could we please do this merge more quickly or could any of you please solve any further merge conflicts?
We had two merge conflicts because there were changes in the main branch before this branch could be merged and I am simply not willing to fix them again if it happens again before merge.
I don't have the energy or time for that, it is super frustrating.
Sorry for this, I hope you understand.
Best regards!
